Port MASTG-TEST-0082: Testing whether the App is Debuggable (ios) (by @appknox) (#3097)
OWASP/owasp-mastg: demos/ios/MASVS-RESILIENCE/MASTG-DEMO-0023/MASTG-DEMO-0023.md

---
platform: ios

> title: Verify App is Debuggable with r2

Reviewer: ```suggestion
title: Debuggable Entitlement Enabled in the entitlements.plist with rabin2
```

code: [swift]
id: MASTG-DEMO-0023
test: MASTG-TEST-0082
---

### Sample

The code snippet below shows sample code that verify the application is debuggable.

> {{ MastgTest.swift }}

Reviewer: Instead of this we need the modified entitlements file, so that we can build the app using it, and the reversed version.

Here's the original: https://github.com/cpholguera/MASTestApp-iOS/blob/main/entitlements.plist

```suggestion
{{ entitlements.plist # entitlements_reversed.plist }}
```

### Steps

1. Unzip the app package and locate the main binary file (@MASTG-TECH-0058), which in this case is `./Payload/MASTestApp.app/MASTestApp`.

> 2. Open the app binary with @MASTG-TOOL-0073 with the `-i` option to run this script.
> 
> {{ debuggable.r2 }}

Reviewer: ```suggestion
2. Run @MASTG-TOOL-0129 with the `-OC` options to obtain the entitlements file.
```

{{ run.sh }}

### Observation

The output reveals the value of the `get-task-allow` entitlement.

{{ output.txt }}

### Evaluation

> The test passes because debugging detection checks are implemented in the app.

Reviewer: ```suggestion
The test fails because the app is debuggable due to the `get-task-allow` entitlement being present.
```
